docs/spec/spec.md

```diff
@@ -0,0 +1,202 @@
+| type\*     | [GroupKind](https://godoc.org/k8s.io/apimachinery/pkg/runtime/schema#GroupKind) | The type of the resource to be provisioned.                                 | Must be Channel. |
+| parameters | runtime.RawExtension (JSON object)                                              | Description of the arguments able to be passed by the provisioned resource. | JSON Schema      |
+
+\*: Required
+
+#### Status
+
+| Field      | Type       | Description                   | Limitations |
+| ---------- | ---------- | ----------------------------- | ----------- |
+| conditions | Conditions | ClusterProvisioner conditions |             |
+
+##### Conditions
+
+- **Ready.**
+
+#### Events
+
+- Resource Created.
+- Resource Removed.
+
+---
+
+## Shared Object Schema
+
+### ProvisionerReference
+
+| Field | Type            | Description | Limitations |
+| ----- | --------------- | ----------- | ----------- |
+| ref\* | ObjectReference |             |             |
+
+\*: Required
+
+### EndpointSpec
+
+| Field                 | Type            | Description | Limitations                |
+| --------------------- | --------------- | ----------- | -------------------------- |
+| targetRef<sup>1</sup> | ObjectReference |             | Must adhere to Targetable. |
+| dnsName<sup>1</sup>   | String          |             |                            |
+
+1: One of (targetRef, dnsName), Required.
+
+### ProvisionedObjectStatus
+
+| Field    | Type   | Description                                                  | Limitations |
+| -------- | ------ | ------------------------------------------------------------ | ----------- |
+| name\*   | String | Name of Object                                               |             |
+| type\*   | String | Fully Qualified Object type.                                 |             |
+| status\* | String | Current relationship between ClusterProvisioner and Object.  |             |
+| reason   | String | Detailed description describing current relationship status. |             |
+
+\*: Required
+
+### Subscribable
+
+| Field       | Type                    | Description                                                           | Limitations                            |
+| ----------- | ----------------------- | --------------------------------------------------------------------- | -------------------------------------- |
+| subscribers | ChannelSubscriberSpec[] | Information about subscriptions used to implement message forwarding. | Filled out by Subscription Controller. |
+
+### ChannelSubscriberSpec
+
+| Field          | Type   | Description                                     | Limitations |
+| -------------- | ------ | ----------------------------------------------- | ----------- |
+| callableDomain | String | The domain name of the endpoint for the call.   |             |
```

Comment:
Let's make these URLs, which will give us more flexibility later for protocol changes, etc if we need it.

Reply:
URI.

Reply:
```
| Field       | Type   | Description                                  | Limitations    |
| ----------- | ------ | -------------------------------------------- | -------------- |
| callableURI | String | The URI name of the endpoint for the call.   | Must be a URL. |
| sinkableURI | String | The URI name of the endpoint for the result. | Must be a URL. |
```